client/utils/events.ts

```diff
@@ -445,6 +446,38 @@ const getDateStringForEvent = (event, dateOnly = false, useLocal = true, withTim
         }
     }
 
+    const isFullDay = event?.dates?.all_day;
+    const noEndTime = event?.dates?.no_end_time;
+
+    const multiDay = !isEventSameDay(start, end);
+
+    if (isFullDay && !multiDay) {
+        return datesToShow = start.format(dateFormat);
+    } else if (noEndTime && !multiDay) {
+        if (withTimezone) {
+            if (!useLocal && (tz || tz == null)) {
+                // eslint-disable-next-line max-len
+                timezoneForEvents = `(${getDateTimeString(start, dateFormat, timeFormat, ' @ ', true, tz ? tz : 'utc')})`;
+            } else {
+                timezoneForEvents = getDateTimeString(start, dateFormat, timeFormat, ' @ ', true);
+            }
+        }
+        return datesToShow = timezoneForEvents;
```

Comment:
not sure in general why there is `return` with `=` a few times, wouldn't just return do?

Reply:
my bad , I forget to remove it.